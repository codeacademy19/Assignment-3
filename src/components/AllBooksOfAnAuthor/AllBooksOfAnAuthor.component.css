.allBooksOfAnAuthor{
   
    flex-wrap: wrap;
    justify-content: space-between;
    margin: 2%;
    margin:2% auto;
    width:80%;
    border:6px solid orange;
    border-radius: 5%; 
}
.allBooksMain{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    margin:0 auto;
    width:100%;
}
.authorName{
    font-size: 2rem;
    color: orange;
    text-align: center;
    background: white;
    padding-left: 2%;
    padding-right: 2%;
    padding-top: 1%;
    padding-bottom: 1%;
    font-style: italic;
}

@media screen and (max-width: 500px) {
    .allBooksOfAnAuthor{
        border:none;
    }
}
