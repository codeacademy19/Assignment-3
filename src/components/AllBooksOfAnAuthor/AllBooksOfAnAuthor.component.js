import React, { Component } from 'react';
import BookPost from '../BookPost/BookPost.component';
import './AllBooksOfAnAuthor.component.css';

class AllBooksOfAnAuthor extends Component {
  render() {
    return (
      <fieldset className="allBooksOfAnAuthor">
        <legend className="authorName">{this.props.author}</legend>
        <div className="allBooksMain">
          {
            this.props.books.map((book, index) => <BookPost book={book} key={book.Name} onLikeClick={() => this.handleLikeClick(index)} />)
            }

        </div>
      </fieldset>
    );
  }
}
export default AllBooksOfAnAuthor;
