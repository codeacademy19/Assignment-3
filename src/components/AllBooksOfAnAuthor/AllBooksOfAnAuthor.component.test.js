import React from 'react';
import renderer from 'react-test-renderer';
import AllBooksOfAnAuthor from './AllBooksOfAnAuthor.component';


describe('AllBooksOfAnAuthor', () => {
  it('should render without crashing', () => {
    const books = [{ Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }, { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }, { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }, { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }];
    const tree = renderer.create(<AllBooksOfAnAuthor author="J.K.ROWLING" books={books} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
