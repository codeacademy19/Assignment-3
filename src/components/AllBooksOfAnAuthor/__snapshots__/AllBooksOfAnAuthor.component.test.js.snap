// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllBooksOfAnAuthor should render without crashing 1`] = `
<fieldset
  className="allBooksOfAnAuthor"
>
  <legend
    className="authorName"
  >
    J.K.ROWLING
  </legend>
  <div
    className="allBooksMain"
  >
    <div
      className="BookPost"
    >
      <div
        className="main"
      >
        <div
          className="backgroundImage"
          style={
            Object {
              "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
            }
          }
        />
        <div
          className="title"
        >
          <span>
            HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
          </span>
        </div>
      </div>
      <div
        className="reaction"
      >
        <span
          className="rating"
        >
          4.5
        </span>
        <img
          alt="heart"
          className="heart"
          src="./Icons/heart-black.svg"
        />
      </div>
    </div>
    <div
      className="BookPost"
    >
      <div
        className="main"
      >
        <div
          className="backgroundImage"
          style={
            Object {
              "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
            }
          }
        />
        <div
          className="title"
        >
          <span>
            HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
          </span>
        </div>
      </div>
      <div
        className="reaction"
      >
        <span
          className="rating"
        >
          4.5
        </span>
        <img
          alt="heart"
          className="heart"
          src="./Icons/heart-black.svg"
        />
      </div>
    </div>
    <div
      className="BookPost"
    >
      <div
        className="main"
      >
        <div
          className="backgroundImage"
          style={
            Object {
              "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
            }
          }
        />
        <div
          className="title"
        >
          <span>
            HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
          </span>
        </div>
      </div>
      <div
        className="reaction"
      >
        <span
          className="rating"
        >
          4.5
        </span>
        <img
          alt="heart"
          className="heart"
          src="./Icons/heart-black.svg"
        />
      </div>
    </div>
    <div
      className="BookPost"
    >
      <div
        className="main"
      >
        <div
          className="backgroundImage"
          style={
            Object {
              "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
            }
          }
        />
        <div
          className="title"
        >
          <span>
            HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
          </span>
        </div>
      </div>
      <div
        className="reaction"
      >
        <span
          className="rating"
        >
          4.5
        </span>
        <img
          alt="heart"
          className="heart"
          src="./Icons/heart-black.svg"
        />
      </div>
    </div>
  </div>
</fieldset>
`;
